pages/_app.js

```diff
@@ -1,20 +1,64 @@
 import React, { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
 
 //default-Slot-Machine:
 export default function App({ Component, pageProps }) {
+  const [storage, setStorage] = useLocalStorageState("storage", {
+    defaultValue: [],
+  });
+
   const [color1, setColor1] = useState("lightgrey");
   const [color2, setColor2] = useState("lightgrey");
   const [color3, setColor3] = useState("lightgrey");
 
   const trafficColor = ["red", "yellow", "green"];
 
+  //"Counting":
+  const [trys, setTrys] = useState(0);
+
   //Win Condition:
   const result =
     color1 === color2 && color1 === color3 && color1 !== "lightgrey"
       ? "won"
       : "lost";
 
+  console.log(storage[0]);
+
+  function save() {
+    //ranking best of 5 mit daten überschreibung:
+    if (storage.length < 5) {
+      setStorage([...storage, trys]);
+      //bis Speicher voll
+    }
+
+    //sortierung des Arrays Storage, absteigend:
+    if (storage.length >= 5 && storage[0] > trys) {
+      //wenn speicher überfüllt, 1. eintrag des arrays löschen, rest klonen:
+      console.log("new record that has to be saved");
+      setStorage([...storage.slice(1, 5), trys]);
+    }
+  }
+
+  function clearStorage() {
+    setStorage([]);
+  }
+
+  // checkVictory and reset game:
+  function checkVictory() {
+    if (color1 === color2 && color1 === color3 && color1 !== "lightgrey") {
```

Comment:
You use the same code here as in line "21", try to use a Function/Outsource to a function (DRY)